File: app/src/main/java/com/example/dars_17_advanced_rv/MainActivity.java
package com.example.dars_17_advanced_rv;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.dars_17_advanced_rv.adapter.NameAdapter;
import com.example.dars_17_advanced_rv.common.Constants;
import com.example.dars_17_advanced_rv.model.NameModel;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private RecyclerView rv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setUpData();
    }

    private void setUpData() {
        rv = findViewById(R.id.rv);
        ArrayList<NameModel> list = genData();
        NameAdapter adapter = new NameAdapter(this, list);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false);
        rv.setLayoutManager(linearLayoutManager);
        rv.setAdapter(adapter);

//        SnapHelper snapHelper = new LinearSnapHelper();
//        snapHelper.attachToRecyclerView(rv);
    }

    private ArrayList<NameModel> genData() {

        ArrayList<NameModel> list = new ArrayList<>(Constants.NAMES.length);
        NameModel model;
        for (String name : Constants.NAMES) {
            model = new NameModel(name);
            list.add(model);
        }
        return list;
    }
}

File: app/src/main/java/com/example/dars_17_advanced_rv/MainActivity.kt
package com.example.dars_17_advanced_rv

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.dars_17_advanced_rv.adapter.NameAdapter
import com.example.dars_17_advanced_rv.common.Commons
import com.example.dars_17_advanced_rv.model.NameModel
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setUpData()
    }

    private fun setUpData() {
        val nameList = genData()
        val nameAdapter = NameAdapter(this, nameList)
        rv.adapter = nameAdapter

    }

    private fun genData(): MutableList<NameModel> {
        val names = Commons.NAMES
        val nameList = mutableListOf<NameModel>()
        names.forEach { nameList += (NameModel(it)) }
        return nameList
    }
}